back-end/app/Http/Controllers/RegistrationController.php

```diff
@@ -66,23 +74,153 @@ class RegistrationController extends Controller
+     *                 type="string",
+     *                 example="user successfully created"
+     *             ),
+     *             @OA\Property(
+     *                 property="token",
+     *                 type="string",
+     *                 example="random string token"
+     *             )
+     *         )
+     *     ),
+     *     @OA\Response(
+     *         response=403,
+     *         description="Email already has an account",
+     *         @OA\JsonContent(
+     *             @OA\Property(
+     *                 property="message",
+     *                 type="string",
+     *                 example="email already has an account"
+     *             )
+     *         )
+     *     ),
+     *     @OA\Response(
+     *         response=400,
+     *         description="Bad Request",
+     *         @OA\JsonContent(
+     *             @OA\Property(
+     *                 property="message",
+     *                 type="string",
+     *                 example="The given data was invalid."
+     *             )
+     *         )
+     *     )
+     * )*/
+    public function adminRegister(Request $request)
     {
         $request->validate([
-            'name' => 'required',
+            'firstName' => 'required|string',
+            'lastName' => 'required|string',
             'email' => 'required|email',
-            'password' => 'required|string'
+            'dateHired' => 'required|date',
+            'role' => 'required|int'
         ]);
 
-        $user = User::create([
-            'name' => $request->name,
+        if (User::where('email', $request->email)->exists()) {
+            return response()->json(['message' => 'email already has a account'], 403);
+        }
+
+        $token = (string) Str::uuid(); // Convert the UUID object to a string
+        // Create a signed URL
+        $signedUrl = URL::temporarySignedRoute(
+            'register.confirm',
+            Carbon::now()->addMinutes(1),
+            ['token' => $token] // Pass the string version of the token
+        );
+
+        // Store the token in the cache
+        Cache::put($token, true, Carbon::now()->addMinutes(30));
+
+        $user  = User::create([
+            'UserFirstName' => $request->firstName,
+            'UserLastName' => $request->lastName,
             'email' => $request->email,
-            'password' => Hash::make($request->password)
+            'DateHired' => $request->dateHired,
+            'UserRoleID' => $request->role,
+            'RegistrationStatus' => 'pending',
+            'RegistrationToken' => $token
+
         ]);
 
+        Mail::to($request->email)->send(new RegisterMail($signedUrl, $request->email));
+
         return response()->json([
-            'message' => 'User created successfully',
-            'token' => $user->createToken('token')->plainTextToken
-        ]);
+            'message' => 'user successfully created',
+            'token' => $token
```

Comment:
```suggestion
            'message' => 'user successfully created'
```

De token terugsturen is onnodig en een beetje "risky"